pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -245,5 +246,9 @@ func (r *DNSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&source.Kind{Type: &clusterv1.ManagedCluster{}},
 			handler.EnqueueRequestsFromMapFunc(clusterEventMapper.MapToDNSPolicy),
 		).
+		// Watches(
```

Comment:
might not do this in this PR but we may want to queue DNSPolicy if changes happen to the DNSRecord (example it gets deleted by accident currently it wont get recreated without some for  of change to the gateway or the policy)

Reply:
Probably want to think about this one. Can see the issue alright, but we ideally don't want to be triggering the policy reconcile unnecessarily i.e. it's just update the record